nep-14.mediawiki

```diff
@@ -65,6 +65,14 @@ Method object has the following structure:
 
 <code>returntype</code> indicates the return type of the method. It can be one of the following values: <code>Signature</code>, <code>Boolean</code>, <code>Integer</code>, <code>Hash160</code>, <code>Hash256</code>, <code>ByteArray</code>, <code>PublicKey</code>, <code>String</code>, <code>Array</code>, <code>Map</code>, <code>InteropInterface</code>, <code>Void</code>.
 
+The method <code>_initialize</code> will be generated if static variables are defined in smart contracts. It will be executed automatically to initialize the static variable BEFORE other methods in contracts are called by using syscall <code>System.Contract.Call</code> and <code>System.Contract.CallEx</code>.
```

Comment:
The method <code>_initialize</code> will be generated if static variables are defined in smart contracts. It will be executed automatically to initialize the static variable BEFORE other methods in contracts are called by using syscall <code>System.Contract.Call</code> and <code>System.Contract.CallEx</code>.
-> 
The method <code>_initialize</code> will be generated if static variables are defined in smart contracts. It will be **automatically executed, during contract registration**, to initialize the static variable BEFORE other methods in contracts are called by using syscall <code>System.Contract.Call</code> and <code>System.Contract.CallEx</code>.

Reply:
@bettybao1209  @shargon, @erikzhang, maybe we need to also explain that when we increase the **version** the **_initialize** will be again called, right?

Reply:
> static variables are defined in smart contracts

I think it's not correct to talk about `static variables` here, it's a different level of abstraction. NEP-14 should only care about `_initialize` existence and calling it when needed, what it does is up to contract.

Reply:
I have made some updates based on your suggestions. Please review it again.